examples/grpc/greeter/client.cpp

```diff
@@ -0,0 +1,84 @@
+#include "greeter.grpc.fb.h"
+#include "greeter_generated.h"
+
+#include <grpc++/grpc++.h>
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+class GreeterClient {
+ public:
+  GreeterClient(std::shared_ptr<grpc::Channel> channel)
+    : stub_(Greeter::NewStub(channel)) {}
+
+  std::string SayHello(const std::string& name) {
+    auto name_offset = mb_.CreateString(name);
+    auto request_offset = CreateHelloRequest(mb_, name_offset);
+    mb_.Finish(request_offset);
+    auto request_msg = mb_.ReleaseMessage<HelloRequest>();
+
+    flatbuffers::grpc::Message<HelloReply> response_msg;
+
+    grpc::ClientContext context;
+
+    auto status = stub_->SayHello(&context, request_msg, &response_msg);
+    if (status.ok()) {
+      const HelloReply* response = response_msg.GetRoot();
+      return response->message()->str();
+    } else {
+      std::cerr << status.error_code() << ": " << status.error_message()
+                << std::endl;
+      return "RPC failed";
+    }
+  }
+
+  template <class F>
```

Comment:
I saw a mixture of inlined and separate-lined templates in `flatbuffers.h` so I thought it was a "whichever fits better on the line" rule, but now that I look I think it's a "inline for methods and separate-line for regular funcs". Is that correct?

Is there any way to express that rule in clang-format? Or do I need to manually format these? Or do you guys use a different formatter? I'm trying my best to infer conventions, but requiring contributors to get formatting details right without automated tooling is kind of asking a lot.